dev/test_lib.py

```diff
@@ -9,6 +9,16 @@
 import cpp.lib  # noqa: E402
 
 
+def test_clang_tidy_conf_merger():
+    orig_checks = "foo-*,bar-pika,-bar-foo"
+    test_func = cpp.lib.ClangTidy.merge_clang_tidy_checks
+
+    assert test_func(orig_checks, "-bar-pika") == "foo-*,-bar-foo,-bar-pika"
+    assert test_func(orig_checks, "-bar-*") == "foo-*,-bar-*"
+    assert test_func(orig_checks, "bar-foo") == "foo-*,bar-pika,bar-foo"
+    assert test_func(orig_checks, "bar-pika") == "foo-*,-bar-foo,bar-pika"
+
+
```

Comment:
The following isn't checked:
* `orig_checks = None` and `new_checks = None` (Likely correct.)
* `orig_checks = ""` and same for `new_checks` (Not immediately obvious to me.)
* `new_checks` never contains a comma, i.e. it's only ever one argument. (Likely correct.)

The tests are good, maybe you can make them a little bit more exhaustive by checking in pairs. Meaning if you check `"-bar-pika"` you also check `"bar-pika"` (already done). If you also do this for `"bar-*"` and `"-bar-foo"`. I think you'll cover all cases.

Feel free to use your own judgement as to which tests are worth adding.